### PR TITLE
Fixed #27418 -- Fixed occasional missing plural forms in JavaScriptCatalog.

### PR DESCRIPTION
**Pull request by wkornewald**
This bug occurs when you have a missing translation in a language with more than two plural forms. In this case, you have a fallback to English and depending on the order in which the catalog is traversed in the for-loop you might end up with an msgid of a fallback English two-plural translation. When that happens, all translations will be reduced to two plural forms.

The unit test only fails about 50% of the time when running against a Django version without the fix due to the bug's random nature.

https://code.djangoproject.com/ticket/27418

### CODE REVIEW

**Comment by claudep** [comment]
Even if not vital to the patch, having plural equations in the po files and a language that really has 3 plural forms would be nice (see http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html).

**Comment by wkornewald** [comment]
I've moved the translations to "ru".

File: tests/view_tests/tests/test_i18n.py [hunk 1]
```diff
@@ -314,6 +314,16 @@ def test_jsi18n_fallback_language(self):
             self.assertContains(response, 'il faut le traduire')
             self.assertNotContains(response, "Untranslated string")
 
+    def test_i18n_fallback_language_plural(self):
+        """
+        Make sure fallback to a language with less plural forms always maintains
```
Comment (timgraham) [inline tests/view_tests/tests/test_i18n.py]: No need for "Make sure" prefix -- just state the expected behavior per [test docstring guidelines](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

File: tests/view_tests/tests/test_i18n.py [hunk 2]
```diff
@@ -314,6 +314,16 @@ def test_jsi18n_fallback_language(self):
...
+        the real language's number of plural forms.
+        """
+        with self.settings(LANGUAGE_CODE='pt'), override('ru'):
+            response = self.client.get('/jsi18n/')
+            catalog = response.context['catalog']
+            self.assertEqual(len(catalog['{count} plural3']), 3)
```
Comment (timgraham) [inline tests/view_tests/tests/test_i18n.py]: `self.assertEqual(catalog['{count} plural3'], ['{count} plural3', '{count} plural3s', '{count} plural3 p3t'])` may help debug the test more easily if it fails.

**Comment by wkornewald** [comment]
I've updated the patch with the suggested changes.

We've never hit the bug before 1.10, but from a quick look at the 1.9.x branch I'd say that older versions should be affected, too. Maybe the ordering of the translations in the .po files plays a more important role here? In any case, I'd rather be on the safe side and not use msgid in the list comprehension, anymore. That just doesn't look right.

**Pull request merged**